shanoir-ng-front/src/app/studies/study/study.component.ts

```diff
@@ -217,6 +234,30 @@ export class StudyComponent extends EntityComponent<Study> {
         return formGroup;
     }
 
+  private getTotalSize(id: number): Promise<number> {
+
+    return this.datasetService.getByStudyId(id).then(datasets => {
+
+      let promises: Promise<any>[] = [];
+
+      if(datasets == null || datasets.length == 0){
+        return 0;
+      }
+
+      let totalSize: number = 0;
+      for (let ds of datasets) {
```

Comment:
Maybe it was how we wanted to design this, but I'm wondering if it's a good idea to make X rest calls for every concerned dataset.
We may make one single call on the study ID and then calculate in the back (?)

The other workaround would be that, if you add size to datasetExpression in the front classes, it will be retrieved automatically with the first call, and you can make the calculation in the front directly.

Reply:
I'm not very satisfied with my design either. I'll discuss it with you this afternoon if you're available.